libcodechecker/analyze/analyzers/analyzer_types.py

```diff
@@ -86,9 +93,37 @@ def check_supported_analyzers(analyzers, context):
         if not analyzer_bin:
             LOG.warning('Failed to detect analyzer binary ' + analyzer_name)
             available_analyzer = False
-        elif not host_check.check_clang(analyzer_bin, check_env):
+        elif not os.path.isabs(analyzer_bin):
+            # If the analyzer is not in an absolute path, try to find it...
+            if analyzer_name == CLANG_SA:
+                found_bin = analyzer_clangsa.ClangSA.\
```

Comment:
Can't we use the "current analyzer" somehow?

It is pretty strange that we have a common base class but instead of virtual calls we are dispatching manually almost always.